src/Model/Model.ts

```diff
@@ -93,18 +93,22 @@ export class Model {
   public static $database: Mongodb;
   public static collectionName?: string;
 
-  protected $collection: Collection;
+  protected $collection: Collection | null = null;
   protected $originalData: any;
   protected $currentData: any;
   protected $isDeleted: boolean;
   protected $options: IModelOptions;
 
-  public constructor(dbObj: Record<string, unknown>, options: IModelOptions) {
-    this.$collection = options.collection;
-    this.$originalData = cloneDeep(dbObj);
-    this.$currentData = dbObj;
+  public constructor(dbObj?: Record<string, unknown>, options?: IModelOptions) {
+    this.$originalData = {};
+    this.$currentData = dbObj === undefined ? {} : dbObj;
```

Comment:
There's something wrong here. If we instantiate from a database query result, $originalData shouldn't be empty because it is used to determine if the object has been modified.
Can you add a test that verifies that after `findOne`, the returned object returns `false` for `isDirty` ?